subsys/usb/device_next/class/bt_hci.c

```diff
@@ -138,7 +138,8 @@ static uint8_t bt_hci_get_bulk_in(struct usbd_class_data *const c_data)
 	struct bt_hci_data *data = usbd_class_get_private(c_data);
 	struct usbd_bt_hci_desc *desc = data->desc;
 
-	if (usbd_bus_speed(uds_ctx) == USBD_SPEED_HS) {
+	if (USBD_SUPPORTS_HIGH_SPEED &&
+	    usbd_bus_speed(uds_ctx) == USBD_SPEED_HS) {
```

Comment:
This does not look good and is easy to forget. I suggest adding a helper macro that could take usbd_bus_speed(), usbd_caps_speed_is_hs() or just speed as argument.

```diff
diff --git a/include/zephyr/usb/usbd.h b/include/zephyr/usb/usbd.h
index 186f79b991b..8ff45bfe678 100644
--- a/include/zephyr/usb/usbd.h
+++ b/include/zephyr/usb/usbd.h
@@ -966,6 +966,15 @@ int usbd_wakeup_request(struct usbd_context *uds_ctx);
  */
 enum usbd_speed usbd_bus_speed(const struct usbd_context *const uds_ctx);
 
+/**
+ * @brief Helper to check if speed is high speed
+ *
+ * @param[in] fspeed Parameter that expands to usbd_speed
+ *
+ * @return 1 if actual speed is high speed
+ */
+#define usbd_speed_is_hs(fspeed) \
+       (USBD_SUPPORTS_HIGH_SPEED && fspeed == USBD_SPEED_HS)

 /**
  * @brief Get highest speed supported by the controller
  *
```

Maybe also shorthand for usbd_bus_speed()/usbd_caps_speed()

```c
+/**
+ * @brief Helper to check if actual device speed is high speed
+ *
+ * @param[in] uds_ctx Pointer to a device context
+ *
+ * @return 1 if actual device speed is high speed
+ */
+#define usbd_bus_speed_is_hs(uds_ctx) \
+       (usbd_speed_is_hs(usbd_bus_speed(uds_ctx)))
```

Reply:
I tend to disagree. Hiding compile time evaluation conditions behind a macro that looks just like a function is IMHO hiding the fact that the code is supposed to go away when compiled with Full-Speed only configuration.

I won't object the macros if you submit the PR yourself, but I just don't think these are justified myself.